bc_obps/registration/api/v2/_user_operators/current.py

```diff
@@ -0,0 +1,50 @@
+from registration.models.operator import Operator
+from service.operator_service_v2 import OperatorServiceV2
+from registration.schema.v2.operator import OperatorIn, OperatorOut
+from registration.models import (
+    UserOperator,
+)
+from service.error_service.custom_codes_4xx import custom_codes_4xx
+from typing import Literal, Tuple
+from django.http import HttpRequest
+from service.data_access_service.user_service import UserDataAccessService
+from registration.api.utils.current_user_utils import get_current_user_guid
+from registration.decorators import authorize, handle_http_errors
+from registration.schema.generic import Message
+from registration.api.router import router
+
+
+# "current" refers to the current user-operator (we look up the current user-operator via the current user)
+# this endpoint retrieves data about both the user-operator and the operator
+
+## GET
+@router.get(
+    "/v2/user-operators/current",
+    response={200: OperatorOut, custom_codes_4xx: Message},
+    tags=["V2"],
+    description="""Retrieves data about the current user-operator and their associated operator.
+    Declined user-operators are excluded from the results.""",
+)
+@authorize(["industry_user"], UserOperator.get_all_industry_user_operator_roles())
+@handle_http_errors()
+def get_current_operator_and_user_operator_v2(request: HttpRequest) -> Tuple[Literal[200], Operator]:
+    user_operator = UserDataAccessService.get_operator_by_user(get_current_user_guid(request))
+    return 200, user_operator
+
+
+## PUT
+@router.put(
+    "/v2/user-operators/current",
+    response={200: OperatorOut, custom_codes_4xx: Message},
+    tags=["V2"],
+    description="""Updates the current user's operator.
+    The endpoint ensures that industry users can only update their own operators.
+    The updated data is saved.""",
+)
+@authorize(["industry_user"], UserOperator.get_all_industry_user_operator_roles())
+@handle_http_errors()
+def update_operator_and_user_operator(
```

Comment:
For this PR, we only need to update the operator, not the user-operator. However, we might need to do the user-operator too from this endpoint at some point (statuses?) so I've kept this in the user-operators route